lib/pure/strutils.nim

```diff
@@ -94,6 +95,18 @@ const
   Letters* = {'A'..'Z', 'a'..'z'}
     ## The set of letters.
 
+  UppercaseLetters* = {'A'..'Z'}
+    ## The set of uppercase ASCII letters.
+
+  LowercaseLetters* = {'a'..'z'}
+    ## The set of lowercase ASCII letters.
+
+  PunctuationChars* = {'!'..'/', ':'..'@', '['..'`', '{'..'~'}
+    ## The set of all ASCII punctuation characters.
+
+  PrintableChars* = Letters + Digits + PunctuationChars + Whitespace
+    ## The set of all printable ASCII characters (letters, digits, whitespace, and punctuation characters).
+
   Digits* = {'0'..'9'}
```

Comment:
This is wrong. The `Digits` constant should be defined before the `PrintableChars` constant. I'm reverting this PR because it breaks CI => https://github.com/nim-lang/Nim/pull/20119/files feel free to make another PR plus a changelog entry.